### PR TITLE
(M) As a visitor on the categories page I want to view tiles of current categories so that I can select what I'm interested in.

### PR DESCRIPTION
**Pull request by bethios**
https://trello.com/c/GMzPLpOD/33-m-as-a-visitor-on-the-categories-page-i-want-to-view-tiles-of-current-categories-so-that-i-can-select-what-i-m-interested-in

https://sacred-art-development.herokuapp.com/categories

screenshot 2017-05-29 15 44 15

### CODE REVIEW

**Comment by bethios** [comment]
[(M) As a visitor on the categories page I want to view tiles of current categories so that I can select what I'm interested in.](https://trello.com/c/GMzPLpOD/33-m-as-a-visitor-on-the-categories-page-i-want-to-view-tiles-of-current-categories-so-that-i-can-select-what-i-m-interested-in)

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -154,24 +156,35 @@ nav{
...
+iframe{
+  border: none;
 }
 
 .button{
   width: 250px;
   margin: 8px;
   font-size: 1.1em;
   background-color: $slate;
+  color: white;
```
Comment (cblant20) [inline app/assets/stylesheets/application.scss:173]: is this the secondary font color? is there somewhere else you can declare it?

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -154,24 +156,35 @@ nav{
...
 
 .footer-col{
   margin: 25px 0 10px 0;
   bottom: 0;
 }
 
-.footer-col:not(:last-child){
-  border-bottom: 1px $accent solid;
+.footer .social {
+  color: white;
```
Comment (cblant20) [inline app/assets/stylesheets/application.scss:183]: all colors should be in variables

File: app/assets/stylesheets/welcome.scss [hunk 3]
```diff
@@ -45,6 +45,65 @@
   font-size: 1.3em;
 }
 
+.social{
+  font-size: 2.85em;
```
Comment (cblant20) [inline app/assets/stylesheets/welcome.scss:49]: variable plz :)

File: app/assets/stylesheets/welcome.scss [hunk 4]
```diff
@@ -45,6 +45,65 @@
...
+  padding: 0 10px;
+  color: $blackish;
+}
+
+.social:hover{
+  color: $accent
```
Comment (cblant20) [inline app/assets/stylesheets/welcome.scss:55]: good naming here

File: app/assets/stylesheets/welcome.scss [hunk 5]
```diff
@@ -45,6 +45,65 @@
...
+}
+
+iframe{
```
Comment (cblant20) [inline app/assets/stylesheets/welcome.scss:58]: eeek iframes are bad. If you can avoid this, avoid this!!!!!

File: app/assets/stylesheets/welcome.scss [hunk 6]
```diff
@@ -45,6 +45,65 @@
...
+.instagram-image-div{
+  height: 40vh;
+  overflow: hidden;
+  margin: 0 auto;
+  text-align: center;
+}
+
+.instagram-comment{
+  padding-top: 15px;
+  font-size: 1.25em;
```
Comment (cblant20) [inline app/assets/stylesheets/welcome.scss:87]: Sass variable plz

File: app/assets/stylesheets/welcome.scss [hunk 7]
```diff
@@ -45,6 +45,65 @@
...
+  max-height: 15vh;
+}
+
+.instagram{
+  height: 60vh;
+}
+
+.categoryLabel{
+  text-align: center;
+  font-size: 1.2em;
```
Comment (cblant20) [inline app/assets/stylesheets/welcome.scss:98]: variable here

File: app/assets/stylesheets/welcome.scss [hunk 8]
```diff
@@ -56,6 +115,13 @@
   }
 
 
+
+}
+
+@media screen and (min-width: 1024px) {
+  .categoryLabel {
+    font-size: 1.75em;
```
Comment (cblant20) [inline app/assets/stylesheets/welcome.scss:123]: variableeee 🐙

File: app/controllers/welcome_controller.rb [hunk 9]
```diff
@@ -1,11 +1,17 @@
...
+
   def index
   end
 
   def about
   end
 
   def new
+    response = HTTParty.get("https://api.instagram.com/v1/users/self/media/recent/?access_token=1501412006.1c07f10.9907313dd1db40ae86c0c9c1f61fe2d0")
+    @instagrams = response["data"]
```
Comment (cblant20) [inline app/controllers/welcome_controller.rb:14]: instagrams plural? is that access token something that needs to be private?

File: app/views/layouts/application.html.erb [hunk 10]
```diff
@@ -26,19 +26,14 @@
...
-    <li class="dropdown-parent nav-li"><%= link_to "ARTISTS" %>
+    <li class="dropdown-parent nav-li"><%= link_to "ARTISTS", categories_path %>
       <div class="dropdown-body">
-        <%= link_to "BEAUTY" %>
-        <%= link_to "CERAMICS AND GLASS" %>
-        <%= link_to "FIBER AND WEARABLES" %>
-        <%= link_to "GRAPHIC ART" %>
-        <%= link_to "HOME" %>
-        <%= link_to "ILLUSTRATION" %>
+        <%= link_to "ORIGINAL ART" %>
```
Comment (cblant20) [inline app/views/layouts/application.html.erb:31]: do these exist?

File: app/views/welcome/_email.html.erb [hunk 11]
```diff
@@ -0,0 +1,5 @@
+<p><strong>GENERAL INQUIRIES:</strong></p>
+<p><%= mail_to('sacredartstore@gmail.com')%></p>
```
Comment (cblant20) [inline app/views/welcome/_email.html.erb:2]: protect this from spam

File: app/views/welcome/_email.html.erb [hunk 12]
```diff
@@ -0,0 +1,5 @@
...
+
+<p><strong>ARTIST SUBMISSIONS:</strong></p>
+<p><%= mail_to('sacredartsubmissions@gmail.com') %></p>
```
Comment (cblant20) [inline app/views/welcome/_email.html.erb:5]: protect this from spam

File: app/views/welcome/_mailinglist.html.erb [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+      </div>
+      <div class="mc-field-group">
+        <label for="mce-LNAME" class="sr-only">Last Name </label>
+        <input type="text" value="" name="LNAME" class="" id="mce-LNAME" placeholder="Last Name">
+      </div>
+      <div id="mce-responses" class="clear">
+        <div class="response" id="mce-error-response" style="display:none"></div>
+        <div class="response" id="mce-success-response" style="display:none"></div>
+      </div>
+      <div style="position: absolute; left: -5000px;" aria-hidden="true"><input type="text" name="b_dd6462c0ac2ea6e4fcc5f8a07_7573d093b0" tabindex="-1" value=""></div>
```
Comment (cblant20) [inline app/views/welcome/_mailinglist.html.erb:21]: what's up with inline styling?

File: app/views/welcome/categories.html.erb [hunk 14]
```diff
@@ -0,0 +1,38 @@
+<div class="row">
+  <div class="col-sm-4">
+    <div class="content-container">
+      <%= image_tag('categories/original.jpg', :alt => 'original art category photo', :class => 'categoryImage') %>
```
Comment (cblant20) [inline app/views/welcome/categories.html.erb]: for accessibility it's better to describe what the photo is. If this is something that will be changed by your client, make sure to include it in the CMS

File: app/views/welcome/categories.html.erb [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+  </div>
+  <div class="col-sm-4">
+    <div class="content-container">
+      <%= image_tag('categories/stationery.jpg', :alt => 'stationery category photo', :class => 'categoryImage') %>
+      <p class="categoryLabel">STATIONERY</p>
+    </div>
+  </div>
+  <div class="col-sm-4">
+    <div class="content-container">
+      <%= image_tag('categories/jewelry.jpg', :alt => 'jewelry category photo', :class => 'categoryImage') %>
```
Comment (cblant20) [inline app/views/welcome/categories.html.erb]: your classes should be named with dashes category-image like standard CSS

File: app/views/welcome/categories.html.erb [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+      <p class="categoryLabel">JEWELRY</p>
+    </div>
+  </div>
+  <div class="col-sm-4">
+    <div class="content-container">
+      <%= image_tag('categories/wearable.jpg', :alt => 'wearables category photo', :class => 'categoryImage') %>
+      <p class="categoryLabel">WEARABLES</p>
```
Comment (cblant20) [inline app/views/welcome/categories.html.erb]: this should be category-label
make sure you are following naming conventions and best practices for the languages

File: app/views/welcome/contact.html.erb [hunk 17]
```diff
@@ -1,8 +1,27 @@
-<div class="row">
-  <div class="col-sm-4 content-container">
-    <iframe src="https://www.google.com/maps/d/u/0/embed?mid=1Mzqwzw1l67n9pskWjsMdyERVkCY" width="100%" height="480"></iframe>
+<div class="row contact-page">
+  <div class="col-sm-6">
+    <div class="text-container contact">
+      <%= render partial: 'welcome/address'%>
+    </div>
+    <div class="content-container">
+      <iframe src="https://www.google.com/maps/d/u/0/embed?mid=1Mzqwzw1l67n9pskWjsMdyERVkCY"  height="480"></iframe>
```
Comment (cblant20) [inline app/views/welcome/contact.html.erb:7]: is this the map that google maps tells you to embed? You shouldn't be using iframes.

File: app/views/welcome/submissions.html.erb [hunk 18]
```diff
@@ -1,2 +1,38 @@
-<h1>Welcome#submissions</h1>
-<p>Find me in app/views/welcome/submissions.html.erb</p>
+<div class="row">
+  <div class="col-sm-3 side-images">
+    <%= image_tag('ceramics.jpg', alt: 'blue and white ceramic vase, small ceramic cups and tea pot', class: 'content-container side-image') %>
+    <%= image_tag('jewelry.jpg', alt: 'constellation necklaces on cards, clock hand earrings on cards', class: 'content-container side-image')%>
```
Comment (cblant20) [inline app/views/welcome/submissions.html.erb:4]: these aren't clickable? why are they here?

File: app/views/welcome/submissions.html.erb [hunk 19]
```diff
@@ -1,2 +1,38 @@
...
+      <p>We curate our collection with a focus on artists who have found their unique voice and who are creating consistent,
+        quality work. Similar to traditional galleries, we work on a consignment basis, which gives us and the artists
+        flexibility to experiment with different pieces and to adjust our collection based on feedback. Our commission is
+        50% of retail sale price, and the artist sets the retail price.</p>
+        <p>We give preference to Chicagoland-based artists, and we work directly with artists, not with agents or representatives.
+        It is helpful if you visit the shop or peruse our website to get a sense of our aesthetic before you submit. We Require:</p>
+
+        <ul>
+          <li><p>Cover letter with bio, artist statement and/or CV</p></li>
+          <li><p>JPGs of your work (at least 3 but as many as 10)</p></li>
```
Comment (cblant20) [inline app/views/welcome/submissions.html.erb:18]: 3-10 images of your work in JPG format

File: app/views/welcome/submissions.html.erb [hunk 20]
```diff
@@ -1,2 +1,38 @@
...
+          <li><p>Link to your online portfolio, website, or online shop</p></li>
```
Comment (cblant20) [inline app/views/welcome/submissions.html.erb:19]: online is repeated

File: app/views/welcome/submissions.html.erb [hunk 21]
```diff
@@ -1,2 +1,38 @@
...
+        <p>Please submit to our dedicated submissions email: <%= mail_to("sacredartsubmissions@gmail.com")%>. Submissions with incomplete
+          information or submissions sent to the incorrect email address will not be considered.</p>
+
+        <p><em>We don’t review artwork on a walk-in basis. You must submit via email for consideration.</em></p>
+
+        <p>You will receive an auto-reply from us confirming that your submission has been received. When we have an
+        opening that is a good fit for you, we will contact you and set up an appointment. If you have not heard from
+        us within a year, please feel free to resubmit with new work. If it has been under a year, you do not need to
+        resubmit to us unless you have a big update to announce – we love to see new websites, online shop launches and
+          new design lines. And by all means please do add us to your e-newsletter contact list!</p>
```
Comment (cblant20) [inline app/views/welcome/submissions.html.erb:33]: remove the "And" from the final sentence

File: spec/features/welcome_spec.rb [hunk 22]
```diff
@@ -71,4 +71,71 @@
     end
   end
 
+  describe "visit CONTACT page" do
+    context "goes to Contact page and shows info" do
```
Comment (cblant20) [inline spec/features/welcome_spec.rb:75]: remove the shows info from the top.

File: spec/features/welcome_spec.rb [hunk 23]
```diff
@@ -71,4 +71,71 @@
...
+      it "shows address" do
+        visit '/contact'
+        page.has_css?('row .col-sm-6 .contact', text: "VISIT")
+      end
+      it "shows map in iframe" do
+        visit '/contact'
+        page.has_css?('row .col-sm-6 iframe', visible: :show)
```
Comment (cblant20) [inline spec/features/welcome_spec.rb:82]: this is not really checking that the iframe is working, just that you have one on the page

File: spec/features/welcome_spec.rb [hunk 24]
```diff
@@ -71,4 +71,71 @@
     end
   end
 
+  describe "visit CONTACT page" do
+    context "goes to Contact page and shows info" do
+      it "shows address" do
+        visit '/contact'
```
Comment (cblant20) [inline spec/features/welcome_spec.rb:77]: this is not checking that the address is there, just VISIT

File: spec/features/welcome_spec.rb [hunk 25]
```diff
@@ -71,4 +71,71 @@
...
+        page.has_css?('row .col-sm-6 .contact', text: "VISIT")
+      end
+      it "shows map in iframe" do
+        visit '/contact'
+        page.has_css?('row .col-sm-6 iframe', visible: :show)
+      end
+      it "shows email address" do
+        visit '/contact'
+        page.has_css?('row .col-sm-6 div', text: "GENERAL INQUIRIES:")
```
Comment (cblant20) [inline spec/features/welcome_spec.rb:86]: also not checking the email address

File: spec/features/welcome_spec.rb [hunk 26]
```diff
@@ -71,4 +71,71 @@
...
+      end
+      it "shows social media icons" do
+        visit '/contact'
+        page.has_css?('row .col-sm-6 div', class: 'social ion-social-twitter')
+        page.has_css?('row .col-sm-6 div', class: 'social ion-social-facebook')
+        page.has_css?('row .col-sm-6 div', class: 'social ion-social-instagram')
+      end
+      it "shows store hours" do
```
Comment (cblant20) [inline spec/features/welcome_spec.rb:94]: also not actually showing the store hours

File: spec/features/welcome_spec.rb [hunk 27]
```diff
@@ -71,4 +71,71 @@
...
+        visit '/contact'
+        page.has_css?('row .col-sm-6 div', text: "GET OUR NEWSLETTER" )
+      end
+    end
+  end
+
+  describe "visit SUBMISSIONS page" do
+    context "goes to submissions page and shows info and pictures" do
+      it "shows images" do
+        visit '/submissions'
```
Comment (cblant20) [inline spec/features/welcome_spec.rb:108]: look for the image itself

File: spec/features/welcome_spec.rb [hunk 28]
```diff
@@ -71,4 +71,71 @@
...
+        page.has_css?('row .col-sm-9 .text-container h1', text: 'ARTIST SUBMISSION GUIDELINES')
+      end
+    end
+  end
+
+  describe "visit NEW page" do
+    context "goes to new page and shows latest instagram" do
+      it "shows image and comment" do
+        visit '/new'
+        page.has_css?('.instagram-image')
```
Comment (cblant20) [inline spec/features/welcome_spec.rb:122]: make sure it's loading more than 1

File: spec/features/welcome_spec.rb [hunk 29]
```diff
@@ -71,4 +71,71 @@
...
+        page.has_css?('.instagram-comment')
+      end
+    end
+  end
+
+  describe "visit CATEGORIES page" do
+    context "goes to categories page and shows tiled categories" do
+      it "shows images" do
```
Comment (cblant20) [inline spec/features/welcome_spec.rb:130]: make sure the category count is correct and that they aren't all the same

**Review by cblant20 — CHANGES_REQUESTED** [review]

**Pull request merged**